### PR TITLE
Refactor revert

### PR DESCRIPTION
**Pull request by VictorKmin**

### CODE REVIEW

File: src/middlewares/module/isModulePresent.middleware.ts [hunk 1]
```diff
@@ -1,22 +1,32 @@
...
 import { moduleService } from '../../services';
 
-export const isModulePresent = async (req: Request, res: Response, next: NextFunction) => {
+export const isModulePresent = async (req: IRequestExtended, res: Response, next: NextFunction) => {
 
-  const { module_id } = req.params;
+    const {module_id} = req.params;
 
-  const module = await moduleService.getModuleByID(module_id);
+    if(!module_id) {
```
Comment (VictorKmin) [inline src/middlewares/module/isModulePresent.middleware.ts:12]: same

File: src/controllers/user/user.controller.ts [hunk 2]
```diff
@@ -1,119 +1,79 @@
...
 import { UploadedFile } from 'express-fileupload';
-import * as fs from 'fs-extra';
-import * as Joi from 'joi';
-import { resolve as resolvePath } from 'path';
-import * as uuid from 'uuid';
-
 import { ResponseStatusCodesEnum } from '../../constants';
-import { ErrorHandler } from '../../errors';
 import { HASH_PASSWORD } from '../../helpers';
+import { updatedUserPhotoMv } from '../../helpers/user/updatedUserPhotoMv.helper';
```
Comment (VictorKmin) [inline src/controllers/user/user.controller.ts:5]: optimize require

File: src/controllers/user/user.controller.ts [hunk 3]
```diff
@@ -1,119 +1,79 @@
...
 
-      fs.mkdirSync(resolvePath(`${appRoot}/static/${photoDir}`), {recursive: true});
-
-      await userPhoto.mv(resolvePath(`${appRoot}/static/${photoDir}/${photoName}`));
-      updateInfo.photo_path = `${photoDir}/${photoName}`;
-    }
+        await userService.updateUser(user_id, updateInfo);
 
-    const updateValidity = Joi.validate(updateInfo, updateDataValidator);
+        const user = await userService.getUserByID(user_id);
```
Comment (VictorKmin) [inline src/controllers/user/user.controller.ts:61]: Will be better to use ```js {new: true} ``` in service (line 59) and don't find user again here. 
But you should remove the password. You can show how we did it in `getUserById` method or use your own logic (delete user, password);

```js 
------>SERVICE<-------

  updateUser(user_id: string, patchObject: Partial<IUser>): Promise<any> {
    const UserModel = model<UserType>(DatabaseTablesEnum.USER_COLLECTION_NAME, UserSchema);

    return UserModel
      .findByIdAndUpdate(user_id, patchObject, {new: true}) as any;
  }



----->CONTROLLER<-----

const updatedUser = await userService.updateUser(user_id, updateInfo);
// REMOVE PASSWORD

res.json({data: updatedUser});
```

File: src/controllers/question/question.controller.ts [hunk 4]
```diff
@@ -1,102 +1,69 @@
 import { NextFunction, Request, Response } from 'express';
-import * as Joi from 'joi';
-
-import { ResponseStatusCodesEnum, UserRoleEnum } from '../../constants';
-import { ErrorHandler, errors } from '../../errors';
-import { IQuestion, IRequestExtended, IUser } from '../../interfaces';
+import { ResponseStatusCodesEnum } from '../../constants';
+import { calculationPageCount } from '../../helpers/course';
+import { checkAdminRole, questionSortingAttributes } from '../../helpers/question';
```
Comment (VictorKmin) [inline src/controllers/question/question.controller.ts:4]: Optimize require

File: src/controllers/lesson/lesson.controller.ts [hunk 5]
```diff
@@ -1,149 +1,113 @@
...
+            limit = 20,
+            offset = 0,
+            sort = '_id',
+            order,
+            ...filter
+        } = req.query;
 
-    res.json({data: lesson});
-  }
+        lessonSortingAttributes(sort, next);
```
Comment (VictorKmin) [inline src/controllers/lesson/lesson.controller.ts:29]: Same

File: src/controllers/lesson/lesson.controller.ts [hunk 6]
```diff
@@ -1,149 +1,113 @@
 import { NextFunction, Request, Response } from 'express';
-import * as Joi from 'joi';
-
 import { ResponseStatusCodesEnum } from '../../constants';
-import { ErrorHandler, errors } from '../../errors';
+import { calculationPageCount } from '../../helpers/course';
+import { lessonSortingAttributes } from '../../helpers/lesson';
```
Comment (VictorKmin) [inline src/controllers/lesson/lesson.controller.ts:4]: Optimize require

File: src/controllers/lesson/lesson.controller.ts [hunk 7]
```diff
@@ -1,149 +1,113 @@
...
-      offset = 0,
-      sort = '_id',
-      order,
-      ...filter
-    } = req.query;
+        res.json({
+            data: {
+                lesson,
+                count: lesson.length,
+                pageCount: calculationPageCount(lesson.length, limit)
```
Comment (VictorKmin) [inline src/controllers/lesson/lesson.controller.ts:37]: Same error with `pageCount`. 
Try to google something `findAndCountAll`
And check al files

File: src/controllers/lesson/lesson.controller.ts [hunk 8]
```diff
@@ -1,149 +1,113 @@
...
-  async createLesson(req: IRequestExtended, res: Response, next: NextFunction) {
-
-    const lessonValue = req.body;
-    const {_id} = req.user as IUser;
+    async createLesson(req: IRequestExtended, res: Response, next: NextFunction) {
 
-    const lessonValidity = Joi.validate(lessonValue, lessonValidator);
+        const lessonValue = req.body;
+        const {_id} = req.user as IUser;
+        await lessonService.createLesson({...lessonValue, user_id: _id});
```
Comment (VictorKmin) [inline src/controllers/lesson/lesson.controller.ts:14]: line

File: src/controllers/lesson/lesson.controller.ts [hunk 9]
```diff
@@ -1,149 +1,113 @@
...
-    res.json({
-      data: {
-        lesson
-      }
-    });
-  }
+        const {lesson_id} = req.params;
 
-  async updateMyLesson(req: Request, res: Response, next: NextFunction) {
+        const updatingData = req.body as Partial<ILesson>;
```
Comment (VictorKmin) [inline src/controllers/lesson/lesson.controller.ts:66]: Too many vertical spaces

File: src/controllers/lesson/lesson.controller.ts [hunk 10]
```diff
@@ -1,149 +1,113 @@
...
-        lesson,
-        count,
-        pageCount
-      }
-    });
-  }
+        const {_id} = req.user as IUser;
 
-  async getMyLesson(req: IRequestExtended, res: Response, next: NextFunction) {
+        const lesson = await lessonService.getMyLesson(_id);
```
Comment (VictorKmin) [inline src/controllers/lesson/lesson.controller.ts:53]: Too many vertical spaces

File: src/controllers/group/group.controller.ts [hunk 11]
```diff
@@ -1,80 +1,63 @@
 import { NextFunction, Response } from 'express';
-import * as Joi from 'joi';
-
-import { ResponseStatusCodesEnum } from '../../constants';
-import { ErrorHandler } from '../../errors';
+import { regexFilterParams } from '../../helpers/group';
```
Comment (VictorKmin) [inline src/controllers/group/group.controller.ts:2]: Line
Optimize require

File: src/controllers/group/group.controller.ts [hunk 12]
```diff
@@ -1,80 +1,63 @@
...
+            pageIndex,
+            offset = pageSize * pageIndex,
+            order = '_id',
+            ...filterParams
+        } = req.query;
+
+        const updatedFilterParams = regexFilterParams(filterParams);
+
+        const groups = await groupService.getAllGroups(updatedFilterParams, +pageSize, offset, order);
+        const count = await groupService.getSizeOfAll(updatedFilterParams) as number;
```
Comment (VictorKmin) [inline src/controllers/group/group.controller.ts:20]: Maybe use something like this method to the `pageCount` comment

File: src/controllers/course/course.controller.ts [hunk 13]
```diff
@@ -1,59 +1,39 @@
...
-    } = req.query;
+        const {limit = 20,
+            offset = 0,
+            sort = '_id',
+            order,
+            ...filter
+        } = req.query;
 
-    const filterValidity = Joi.validate(filter, CoursefilterParametresValidator);
+        courseSortingAttributes(sort, next);
```
Comment (VictorKmin) [inline src/controllers/course/course.controller.ts:17]: Same. Just throw an error inside

File: src/controllers/module/module.controller.ts [hunk 14]
```diff
@@ -1,59 +1,41 @@
...
+        const {
+            limit = 20,
+            offset = 0,
+            sort = '_id',
+            order,
+            ...filter
+        } = req.query;
 
-    const filterValidity = Joi.validate(filter, moduleFilterValitator);
+        moduleSortingAttributes(sort, next);
```
[the remark this answers is not in the document]
Comment (OlehKosylo) [reply src/controllers/module/module.controller.ts:19]: Done

File: src/controllers/module/module.controller.ts [hunk 15]
```diff
@@ -1,59 +1,41 @@
 import { NextFunction, Response } from 'express';
-import * as Joi from 'joi';
-
-import { ResponseStatusCodesEnum } from '../../constants';
-import { ErrorHandler } from '../../errors';
-import { IModule, IRequestExtended } from '../../interfaces';
+import { calculationPageCount } from '../../helpers/course';
+import { moduleSortingAttributes } from '../../helpers/module';
```
[the remark this answers is not in the document]
Comment (OlehKosylo) [reply src/controllers/module/module.controller.ts:3]: Done

File: src/controllers/module/module.controller.ts [hunk 16]
```diff
@@ -1,59 +1,41 @@
 import { NextFunction, Response } from 'express';
-import * as Joi from 'joi';
-
-import { ResponseStatusCodesEnum } from '../../constants';
-import { ErrorHandler } from '../../errors';
-import { IModule, IRequestExtended } from '../../interfaces';
+import { calculationPageCount } from '../../helpers/course';
```
[the remark this answers is not in the document]
Comment (OlehKosylo) [reply src/controllers/module/module.controller.ts:2]: Done

File: src/controllers/question/question.controller.ts [hunk 17]
```diff
@@ -1,102 +1,69 @@
...
+
+        const {
+            limit = 20,
+            offset = 0,
+            sort = '_id',
+            order,
+            ...filter
+        } = req.query;
+
+        questionSortingAttributes(sort, next);
```
[the remark this answers is not in the document]
Comment (OlehKosylo) [reply src/controllers/question/question.controller.ts:20]: Done

File: src/controllers/question/question.controller.ts [hunk 18]
```diff
@@ -1,102 +1,69 @@
...
+        const questions = await questionService.getQuestions(+limit, +offset, sort, order, filter);
+
+        res.json({
+            data: {
+                questions,
+                count: questions.length,
+                pageCount: calculationPageCount(questions.length, limit)
```
[the remark this answers is not in the document]
Comment (OlehKosylo) [reply src/controllers/question/question.controller.ts:27]: I don't understand why, this logic was written before me.

File: src/controllers/lesson/lesson.controller.ts (hunk 9 above)
Comment (OlehKosylo) [reply src/controllers/lesson/lesson.controller.ts:66]: Done

File: src/controllers/lesson/lesson.controller.ts (hunk 10 above)
Comment (OlehKosylo) [reply src/controllers/lesson/lesson.controller.ts:53]: Done

File: src/controllers/group/group.controller.ts (hunk 11 above)
Comment (OlehKosylo) [reply src/controllers/group/group.controller.ts:2]: Done

File: src/controllers/course/course.controller.ts (hunk 13 above)
Comment (OlehKosylo) [reply src/controllers/course/course.controller.ts:17]: Done

File: src/controllers/group/group.controller.ts (hunk 12 above)
Comment (OlehKosylo) [reply src/controllers/group/group.controller.ts:20]: In method to the pageCount we make simple calculations, in this case, we got info of database.
You want a method, which will wrap both of them? (getAllGroup, getSizeOfAll)

File: src/controllers/question/question.controller.ts (hunk 18 above)
[the remark this answers is not in the document]
Comment (OlehKosylo) [reply src/controllers/question/question.controller.ts:27]: We can write limit = 1000 records in the request

[the remark this answers is not in the document]
Comment (VictorKmin) [reply src/controllers/question/question.controller.ts:31]: Discussed in telegram

File: src/controllers/group/group.controller.ts (hunk 12 above)
Comment (VictorKmin) [reply src/controllers/group/group.controller.ts:20]: Discussed in telegram

File: src/controllers/question/question.controller.ts [hunk 19]
```diff
@@ -1,102 +1,69 @@
...
-  async getMyQuestions(req: IRequestExtended, res: Response, next: NextFunction) {
+    async getMyQuestions(req: IRequestExtended, res: Response, next: NextFunction) {
 
-    const {_id} = req.user as IUser;
-    const {limit = 20, offset = 20} = req.query;
-    const questions = await questionService.getMyQuestion(_id, +limit, +offset);
-    const count = questions.length;
-    const pageCount = Math.ceil(count / limit);
+        const {_id} = req.user as IUser;
+        const {limit = 20, offset = 20} = req.query;
```
[the remark this answers is not in the document]
Comment (OlehKosylo) [reply src/controllers/question/question.controller.ts:51]: Done

File: src/controllers/user/user.controller.ts (hunk 3 above)
Comment (OlehKosylo) [reply src/controllers/user/user.controller.ts:56]: Done

File: src/controllers/lesson/lesson.controller.ts (hunk 8 above)
Comment (OlehKosylo) [reply src/controllers/lesson/lesson.controller.ts:14]: Done

**Pull request merged**